lib/src/model.dart

```diff
@@ -1896,25 +2110,25 @@ abstract class ModelElement extends Nameable
   // somehow.
   ModelElement(this._element, this._library);
 
+  // TODO(jcollins-g): this way of using the optional parameter is messy,
+  // clean that up.
+  // TODO(jcollins-g): Refactor this into class-specific factories that
+  // call this one.
+  // TODO(jcollins-g): enforce this.
   /// Do not construct any ModelElements unless they are from this constructor.
-  /// TODO(jcollins-g): enforce this.
   /// Specify enclosingClass only if this is to be an inherited object.
   /// Specify index only if this is to be an EnumField.forConstant.
-  /// Specify enclosingCombo (a GetterSetterCombo) only if this is to be an
-  /// Accessor.
-  /// TODO(jcollins-g): this way of using the optional parameter is messy,
-  /// clean that up.
   factory ModelElement.from(Element e, Library library,
-      {Class enclosingClass, int index, ModelElement enclosingCombo}) {
+      {Class enclosingClass, int index, Accessor getter, Accessor setter}) {
```

Comment:
Just an idea: Some of the code might be cleaner if you allowed `e` to be `null` and simply returned `null`. It would eliminate the need for a fair number of null checks. It's also consistent with some of the other factories.